apps/yggdrasil/test/yggdrasil/character_test.exs

```diff
@@ -0,0 +1,156 @@
+defmodule Yggdrasil.CharacterTest do
+  use ExUnit.Case, async: false
+
+  alias Yggdrasil.{Repo, Game, User, Character}
+
+  @min_len 3
+
+  @user1   %{username: "tester",
+             password: "test123", 
+             password_confirmation: "test123" }
+
+  @user2   %{username: "tester2",
+             password: "test123", 
+             password_confirmation: "test123" }
+
+  @game1   %{name: "abcdef", description: "abcdefghi"}
+  @game2   %{name: "ghijkl", description: "abcdefghi"}
+
+  @character %{name: "foobar", user_id: nil, game_id: nil}
+  @invalid_character %{name: "ab", user_id: nil, game_id: nil}
+
+  @invalid_length_msg "should be at least %{count} character(s)"
+  @missing_assoc_msg "does not exist"
+  @unique_msg "has already been taken"
+  @required_msg "can't be blank"
+
+  setup tags do
+    unless tags[:async] do
+      Ecto.Adapters.SQL.restart_test_transaction(Yggdrasil.Repo, [])
+    end
+
+    :ok
+  end
+
+  setup _context do
+    {:ok, user1} = %User{}
+    |> User.create_changeset(@user1)
+    |> Repo.insert
+
+    {:ok, user2} = %User{}
+    |> User.create_changeset(@user2)
+    |> Repo.insert
+
+    {:ok, game1} = %Game{}
+    |> Game.changeset(@game1)
+    |> Repo.insert
+
+    {:ok, game2} = %Game{}
+    |> Game.changeset(@game2)
+    |> Repo.insert
+
+    {:ok, %{user1: user1, user2: user2, game1: game1, game2: game2}}
+  end
+
+  test "valid character produces a valid changeset", ctx do
+    user = ctx.user1
+    game = ctx.game1
+
+    valid_character = %{@character | user_id: user.id, game_id: game.id}
+
+    character = Character.changeset(%Character{}, valid_character)
+    assert character.valid? == true
+  end
+
+  test "valid character changset with valid references inserts", ctx do
+    user = ctx.user1
+    game = ctx.game1
+
+    valid_character = %{@character | user_id: user.id, game_id: game.id}
+
+    character = Character.changeset(%Character{}, valid_character)
+    assert character.valid? == true
+
+    {:ok, _character} = Repo.insert character
+  end
+
+  test "invalid character name length < #{@min_len}" do
```

Comment:
Yeah, true. Didn't enforce one, thus no test.

Can certainly enforce it on the changeset, not sure there is need to have postgres do this - from what I have read it seems to be very efficient with variable length columns so I believe ours are string? Unless we just need a hard stop there.

I'd imagine maybe 50 characters or less is probably enough don't know need to google that I guess.